File: Flow-filler.py
def get_variable_types():
  """Gets the variable types from the user."""
  variable_types = []
  while True:
    variable_type = input("Enter the variable type (int, float, str, bool): ")
    if variable_type in ["int", "float", "str", "bool"]:
      variable_types.append(variable_type)
    else:
      print("Invalid variable type.")
      continue
    if input("Do you want to add another variable type? (y/n): ") == "n":
      break
  return variable_types

def get_variable_values(variable_types):
  """Gets the variable values from the user."""
  variable_values = []
  for variable_type in variable_types:
    variable_value = input(f"Enter the value for the {variable_type} variable: ")
    variable_values.append(variable_value)
  return variable_values

def populate_flowchart(variable_types, variable_values):
  """Populates the flowchart with the variable types and values."""
  flowchart = []
  for i in range(len(variable_types)):
    variable_type = variable_types[i]
    variable_value = variable_values[i]
    flowchart.append(f"{variable_type}: {variable_value}")
  return flowchart

def main():
  """The main function."""
  variable_types = get_variable_types()
  variable_values = get_variable_values(variable_types)
  flowchart = populate_flowchart(variable_types, variable_values)
  print(flowchart)

if __name__ == "__main__":
  main()

File: Flow_filler.js
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static List<String> getVariableTypes() {
        List<String> variableTypes = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);
        while (true) {
            System.out.println("Enter the variable type (int, float, str, bool): ");
            String variableType = scanner.nextLine();
            if (variableType.equals("int") || variableType.equals("float") || variableType.equals("str") || variableType.equals("bool")) {
                variableTypes.add(variableType);
            } else {
                System.out.println("Invalid variable type.");
                continue;
            }
            System.out.println("Do you want to add another variable type? (y/n): ");
            if (scanner.nextLine().equals("n")) {
                break;
            }
        }
        return variableTypes;
    }

    public static List<String> getVariableValues(List<String> variableTypes) {
        List<String> variableValues = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);
        for (String variableType : variableTypes) {
            System.out.printf("Enter the value for the %s variable: \n", variableType);
            String variableValue = scanner.nextLine();
            variableValues.add(variableValue);
        }
        return variableValues;
    }

    public static List<String> populateFlowchart(List<String> variableTypes, List<String> variableValues) {
        List<String> flowchart = new ArrayList<>();
        for (int i = 0; i < variableTypes.size(); i++) {
            String variableType = variableTypes.get(i);
            String variableValue = variableValues.get(i);
            flowchart.add(variableType + ": " + variableValue);
        }
        return flowchart;
    }

    public static void main(String[] args) {
        List<String> variableTypes = getVariableTypes();
        List<String> variableValues = getVariableValues(variableTypes);
        List<String> flowchart = populateFlowchart(variableTypes, variableValues);
        System.out.println(flowchart);
    }
}

